### PR TITLE
feat: ResolveAll endpoint for bulk evaluation

### PR DESCRIPTION
**Pull request by james-milligan**
## This PR

- introduces the resolve all endpoint for bulk flag evaluation

### Related Issues

https://github.com/open-feature/flagd/issues/234

### CODE REVIEW

**Comment by james-milligan** [comment]
LINTING FIXES:
Added the `//nolint:funlen` annotation to the Notify function of `Kubernetes_sync.go`. To drop the functions lines by 2 we will need to abstract out functionality that is used once, or drop comments. Both options will make the code more confusing for developers.

File: pkg/eval/json_evaluator.go [hunk 1]
```diff
@@ -103,6 +103,68 @@ func resolve[T constraints](reqID string, key string, context *structpb.Struct,
...
+				context,
+				je.evaluateVariant,
+				je.state.Flags[flagKey].Variants,
+			)
+			if err != nil {
+				je.Logger.WarnWithID(reqID, fmt.Sprintf("Bulk evaluation: key %s returned error %s", flagKey, err.Error()))
+				continue
+			}
+			values = append(values, NewAnyValue(value, variant, reason, flagKey))
+		case string:
```
Comment (skyerus) [inline pkg/eval/json_evaluator.go:124]: Can we simplify these cases somehow to avoid duplicating the logic?

**Comment by toddbaert** [comment]
I tested this and it works as expected. I agree with @skyerus that it might be nice to reduce some duplication, but that may not be possible.

**Review by skyerus — APPROVED** [review]

File: pkg/eval/json_evaluator.go [hunk 2]
```diff
@@ -103,6 +103,57 @@ func resolve[T constraints](reqID string, key string, context *structpb.Struct,
...
+			value, variant, reason, err = resolve[map[string]any](
+				reqID,
+				flagKey,
+				context,
+				je.evaluateVariant,
+				je.state.Flags[flagKey].Variants,
+			)
+		}
+		if err != nil {
+			je.Logger.WarnWithID(reqID, fmt.Sprintf("Bulk evaluation: key %s returned error %s", flagKey, err.Error()))
```
Comment (skyerus) [inline pkg/eval/json_evaluator.go:149]: Why is this only a warn?

File: pkg/service/connect_service.go [hunk 3]
```diff
@@ -138,6 +138,61 @@ func (s *ConnectService) setupServer() (net.Listener, error) {
...
+				Reason:  value.Reason,
+				Variant: value.Variant,
+				Value: &schemaV1.AnyFlag_DoubleValue{
+					DoubleValue: v,
+				},
+			}
+		case map[string]any:
+			val, err := structpb.NewStruct(v)
+			if err != nil {
+				s.Logger.WarnWithID(reqID, fmt.Sprintf("struct response construction: %v", err))
```
Comment (skyerus) [inline pkg/service/connect_service.go:180]: Why not error level?

File: pkg/eval/json_evaluator.go (hunk 2 above)
Comment (james-milligan) [reply pkg/eval/json_evaluator.go]: resolved

**Review by toddbaert — APPROVED** [review]

**Pull request merged**